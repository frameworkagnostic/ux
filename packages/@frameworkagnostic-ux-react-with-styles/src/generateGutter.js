
import { GUTTERS } from './constants';

export function generateGutter(breakpoint) {
  const styles = {};

  GUTTERS.forEach((spacing, index) => {
    if (index === 0) {
      // Skip the default style.
      return;
    }

    styles[`spacing-${breakpoint}-${spacing}`] = {
      margin: -spacing / 2,
      width: `calc(100% + ${spacing}px)`,
      '& > $item': {
        padding: spacing / 2,
      },
    };
  });

  return styles;
}
