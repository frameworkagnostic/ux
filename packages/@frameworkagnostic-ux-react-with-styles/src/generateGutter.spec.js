
import { generateGutter } from './generateGutter';
import { breakpointXsStyles } from '../__mocks__/generate-gutter.mock';

test('generateGutter xs', () => {
  const styles = generateGutter('xs');

  expect(styles).toEqual(
    expect.objectContaining([
      "spacing-xs-8",
      "spacing-xs-16",
      "spacing-xs-24",
      "spacing-xs-40",
    ].reduce((acc, key) => {
      acc[key] = {
        margin: expect.any(Number),
        width: expect.any(String),
        "& > $item": {
          padding: expect.any(Number)
        },
      };
      return acc;
    }, {})),
  );

  expect(styles).toEqual(
    expect.objectContaining(Object.keys(styles).reduce((acc, key) => {
      acc[key] = breakpointXsStyles[key];
      return acc;
    }, {})),
  );

  expect(Object.keys(breakpointXsStyles).sort()).toEqual(
    expect.arrayContaining(Object.keys(styles).sort()),
  );

});
