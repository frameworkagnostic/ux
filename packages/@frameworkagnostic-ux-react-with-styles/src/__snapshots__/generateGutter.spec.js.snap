// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateGutter xs 1`] = `
Object {
  "spacing-xs-16": Object {
    "& > $item": Object {
      "padding": 8,
    },
    "margin": -8,
    "width": "calc(100% + 16px)",
  },
  "spacing-xs-24": Object {
    "& > $item": Object {
      "padding": 12,
    },
    "margin": -12,
    "width": "calc(100% + 24px)",
  },
  "spacing-xs-32": Object {
    "& > $item": Object {
      "padding": 16,
    },
    "margin": -16,
    "width": "calc(100% + 32px)",
  },
  "spacing-xs-40": Object {
    "& > $item": Object {
      "padding": 20,
    },
    "margin": -20,
    "width": "calc(100% + 40px)",
  },
  "spacing-xs-8": Object {
    "& > $item": Object {
      "padding": 4,
    },
    "margin": -4,
    "width": "calc(100% + 8px)",
  },
}
`;
